### PR TITLE
Some doorkeeper proposals

### PR DESCRIPTION
**Pull request by gravitystorm**
Here's some proposed changes to your doorkeeper pull request. I think it's easier to open them as a PR against the branch on your repo, or at least I can't think of a better way.

They cover a few bootstrap / forms / i18n topics but there's nothing major here.

### CODE REVIEW

File: app/views/oauth2_applications/index.html.erb [hunk 1]
```diff
@@ -2,12 +2,22 @@
...
+      <th><%= t ".permissions" %></th>
+      <th></th>
+      <th></th>
+    </thead>
+    <tbody>
+      <%= render :partial => "application", :collection => @applications %>
+    </tbody>
+  </table>
+<% else %>
+  <p><%= t ".no_applications_html", :oauth2 => link_to(t(".oauth_2"), "https://oauth.net/2/") %></p>
```
Comment (tomhughes) [inline app/views/oauth2_applications/index.html.erb:18]: What's the reason for pulling out the link like that? It creates a potential lego issue in the translations and the main translation is already an HTML string anyway - is it just to stop translators changing the link target?

File: config/locales/en.yml [hunk 2]
```diff
@@ -143,6 +146,9 @@ en:
         pass_crypt: "Password"
         pass_crypt_confirmation: "Confirm Password"
     help:
+      doorkeeper/application:
+        confidential: "Application will be used where the client secret can be kept confidential, eg mobile apps and single page apps"
```
Comment (tomhughes) [inline config/locales/en.yml]: I suspect you moved this and then rebased on a newer version of my tree where I had edited the original although your version doesn't look exactly like the original upstream version either as it seems to have the logic backwards?

File: config/locales/en.yml [hunk 3]
```diff
@@ -485,6 +491,11 @@ en:
       comment: Comment
       newer_comments: "Newer Comments"
       older_comments: "Older Comments"
+  doorkeeper:
+    flash:
+      applications:
+        create:
+          notice: Application Registered.
```
Comment (tomhughes) [inline config/locales/en.yml:498]: Was there a reason for pulling this in from the upstream translations? I don't think we use it directly do we, though the doorkeeper controllers that we inherit from do.

File: app/views/oauth2_applications/index.html.erb (hunk 1 above)
Comment (gravitystorm) [reply app/views/oauth2_applications/index.html.erb:18]: It was done this way just to match the corresponding `no_apps_html` from the OAuth (1) client applications page.

File: config/locales/en.yml (hunk 2 above)
Comment (gravitystorm) [reply config/locales/en.yml]: Oops, I didn't notice the change in the text. Good catch. I'll update this shortly.

File: config/locales/en.yml (hunk 3 above)
Comment (gravitystorm) [reply config/locales/en.yml:498]: It's part of the created vs registered thing - I wanted to keep the verb consistent between the page titles, action buttons and flash messages.

File: app/views/oauth2_applications/index.html.erb (hunk 1 above)
Comment (tomhughes) [reply app/views/oauth2_applications/index.html.erb:18]: Ah ok. I just realised there are two versions of `no_applications_html` but only one of `oauth_2` so I thought you might be sharing that but it seems not so I'm wondering if the other one works properly?

**Pull request merged**